src/games/supported/Freeway.hpp

```diff
@@ -66,11 +66,24 @@ class FreewaySettings : public RomSettings {
 
         virtual int lives() { return 0; }
 
-    private:
+        // returns a list of mode that the game can be played in
+        // in this game, there are 8 available modes
+        ModeVect getAvailableModes();
+
+        // set the mode of the game
+        // the given mode must be one returned by the previous function
+        void setMode(game_mode_t, System &system, StellaEnvironment& environment);
 
+        // returns a list of difficulties that the game can be played in
+        // in this game, there are 2 available difficulties
+        DifficultyVect getAvailableDifficulties();
+
+    private:
         bool m_terminal;
         reward_t m_reward;
         reward_t m_score;
+        game_mode_t m_mode;
+        static const unsigned int m_num_modes = 8;

```

Comment:
Can you use a const method instead, numModes(), to match rom()?

Reply:
Done.